static/code/objectPanel.js

```diff
@@ -0,0 +1,218 @@
+ */
+Code.ObjectPanel.addLink = function(prop) {
+  var name = prop.name;
+  var type = prop.type;
+  var newParts = Code.ObjectPanel.parts.concat(name);
+  var selector = Code.Common.partsToSelector(newParts);
+  var a = document.createElement('a');
+  a.href = '/code?' + encodeURI(selector);
+  a.target = '_blank';
+  a.setAttribute('data-link', name);
+  a.addEventListener('click', Code.ObjectPanel.click);
+  var typeSymbol = Code.ObjectPanel.TYPES[type];
+  if (typeSymbol) {
+    var span = document.createElement('span');
+    span.className = 'objectType';
+    span.appendChild(document.createTextNode(typeSymbol));
+    span.title = type;
+    a.appendChild(span);
+  }
+  var div = document.createElement('div');
+  div.appendChild(document.createTextNode(name));
+  a.appendChild(div);
+  results.appendChild(a);
+};
+
+/**
+ * Symbols to print next to properties.
+ */
+Code.ObjectPanel.TYPES = {
+  'object': '{}',
+  'symbol': '☆',
+  'function': '𝑓'
+};
+
+/**
+ * When a property is clicked, trigger an update.
+ * @param {!Event} e Click event.
+ */
+Code.ObjectPanel.click = function(e) {
+  var name = e.currentTarget.getAttribute('data-link');
+  var newParts = Code.ObjectPanel.parts.concat(name);
+  var selector = Code.Common.partsToSelector(newParts);
+  // Store the new selector in sessionStorage for all frames to see.
+  sessionStorage.setItem(Code.Common.SELECTOR, selector);
+  // Alert the top "/code" frameset that there's been a selector change.
+  window.parent.parent.postMessage('ping', '*');
+  // Don't navigate to this link.
+  e.preventDefault();
+};
+
+/**
+ * Currently selected property, or null if none.
+ * @type {?Element}
+ */
+Code.ObjectPanel.highlighted = null;
+
+/**
+ * Highlight the currently selected property (if any).
+ * The current selection is based on a value in sessionStorage.
+ */
+Code.ObjectPanel.highlight = function() {
+  var selector = sessionStorage.getItem(Code.Common.SELECTOR);
+  var parts = Code.Common.selectorToParts(selector);
+  var part = parts ? parts[Code.ObjectPanel.parts.length] : null;
+  var newHighlighted = null;
+  for (var i = 0, link; (link = results.childNodes[i]); i++) {
+    if (link.getAttribute('data-link') === part) {
+      newHighlighted = link;
+    }
+  }
+  if (newHighlighted !== Code.ObjectPanel.highlighted) {
+    if (Code.ObjectPanel.highlighted) {
+      Code.ObjectPanel.highlighted.className = '';
+    }
+    if (newHighlighted) {
+      newHighlighted.className = 'highlighted';
+      if (newHighlighted.scrollIntoView) {
+        newHighlighted.scrollIntoView({block: 'nearest', inline: 'nearest'});
+      }
+    }
+    Code.ObjectPanel.highlighted = newHighlighted;
+  }
+};
+
+/**
+ * Remove any properties that are shadowed by objects higher on the inheritance
+ * chain.  Also sort the properties alphabetically.
+ * @param {Array<!Array<string>>} data Property names from Code City.
+ */
+Code.ObjectPanel.filterShadowed = function(data) {
```

Comment:
Why duplicate this?

Reply:
This one filters objects, the other one filters strings.